cmd/cortex/run.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+if [ -n "${CORTEX_ULIMIT_NOFILES:-}" ]; then
+    current_limit=$(ulimit -Hn)
+    if [ "$current_limit" != "unlimited" ]; then
+        if [ $CORTEX_ULIMIT_NOFILES -gt $current_limit ]; then
+            echo "Setting file description limit to $CORTEX_ULIMIT_NOFILES"
+            ulimit -Hn $CORTEX_ULIMIT_NOFILES
```

Comment:
I understand modifying hard limits inside the container might solve the problem for some environments in which containers can modify the kernel limits. But I am not sure we should encourage that way of doing things.

I think we should document how to modify this in dockerd/containerd/k8s properly. It's not easy to do, as you have correctly reported.